### PR TITLE
Add some unit tests for ASCIIEncoding

### PR DESCRIPTION
**Pull request by GrabYourPitchforks**
No runtime changes. Just adds some missing test cases for the `ASCIIEncoding` class.

### CODE REVIEW

File: src/System.Text.Encoding/tests/ASCIIEncoding/ASCIIEncodingDecode.netcoreapp.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        [InlineData("hello\u1234there!", 16)]
+        [InlineData("\ud800\udc00", 10)]
+        public void GetByteCount_WithReplacementFallback(string input, int expectedByteCount)
+        {
+            Encoding encoding = Encoding.GetEncoding("ascii", new EncoderReplacementFallback("abcde"), DecoderFallback.ExceptionFallback);
+            Assert.Equal(expectedByteCount, encoding.GetByteCount(input));
+        }
+
+        [Fact]
+        public void GetByteCount_WithSingleCharNonAsciiReplacementFallback_ValidatesAscii()
```
Comment (GrabYourPitchforks) [inline src/System.Text.Encoding/tests/ASCIIEncoding/ASCIIEncodingDecode.netcoreapp.cs]: Note to reviewers: This test case currently fails due to a bug in the current implementation of `ASCIIEncoding`. The actual behavior is that `GetByteCount` returns 1, but correct behavior would instead be for it to throw `ArgumentException`.

I'm working on correcting this issue at the same time that the other `ASCIIEncoding` changes come online.

**Comment by karelz** [comment]
@GrabYourPitchforks what's the status? I don't see update in 2 weeks ... if you're not ready to push it forward, maybe we can close it and reopen when you're ready to give it the time it needs?

**Comment by GrabYourPitchforks** [comment]
@karelz The corresponding coreclr change is held up in PR review. This corefx PR review is signed off; I'm just waiting for the coreclr change to go through first so I don't break the build.

**Comment by karelz** [comment]
OK, if it is 1-2 weeks away, I am fine keeping it open. Otherwise I would recommend to close it, until there is the right time for it ;)
Thanks!

**Comment by karelz** [comment]
@GrabYourPitchforks do you have better ETA?

**Comment by GrabYourPitchforks** [comment]
This went in as part of https://github.com/dotnet/corefx/commit/d817d68fc6e2fc78624540aa90a060f4d5bf0eee.

**Pull request closed without merging**